ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -519,59 +542,75 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
-        includesAllParticipants: this.state.includesAllParticipants,
-        cohortIds: this.state.selectedCohortIds,
-        prePackagedConceptSet: this.getPrePackagedConceptSet(),
-        values: this.state.selectedValues
-      };
-      try {
-        const dataSetPreviewResp = await dataSetApi().previewQuery(namespace, id, request);
-        this.setState({
-          previewList: dataSetPreviewResp.domainValue,
-          selectedPreviewDomain: dataSetPreviewResp.domainValue[0].domain
+      const domains = fp.uniq(this.state.selectedValues.map(domainValue => domainValue.domain));
+      const newPreviewList: Map<Domain, DataSetPreviewInfo> = new Map();
+      domains.forEach(domain => {
+        newPreviewList.set(domain, {
+          isLoading: true,
+          values: []
         });
-      } catch (ex) {
-        const exceptionResponse = await ex.json() as unknown as ErrorResponse;
-        let errorText: string;
-        switch (exceptionResponse.statusCode) {
-          case 400:
-            if (exceptionResponse.message ===
-              'Data Sets must include at least one cohort and concept.') {
-              errorText = exceptionResponse.message;
-            } else if (exceptionResponse.message ===
-              'Concept Sets must contain at least one concept') {
-              errorText = 'One or more of your concept sets has no concepts. ' +
-                'Please check your concept sets to ensure all concept sets have concepts.';
-            }
-            break;
-          case 404:
-            if (exceptionResponse.message.startsWith(
-              'Not Found: No Cohort definition matching cohortId')) {
-              errorText = 'Error with one or more cohorts in the data set. ' +
-                'Please submit a bug using the contact support button';
-            }
-            break;
-          case 504:
-            if (exceptionResponse.message ===
-              'Timeout while querying the CDR to pull preview information.') {
-              errorText = 'Query to load data from the All of Us Database timed out. ' +
-                'Please either try again or export data set to a notebook to try there';
-            }
-            break;
-          default:
-            errorText = 'An unexpected error has occurred. ' +
-              'Please submit a bug using the contact support button';
+      });
+      this.setState({
+        previewList: newPreviewList,
+        selectedPreviewDomain: domains[0]
+      });
+      domains.forEach(async domain => {
+        const request: DataSetPreviewRequest = {
+          domain: domain,
+          conceptSetIds: this.state.selectedConceptSetIds,
+          includesAllParticipants: this.state.includesAllParticipants,
+          cohortIds: this.state.selectedCohortIds,
+          prePackagedConceptSet: this.getPrePackagedConceptSet(),
+          values: this.state.selectedValues.map(domainValue => domainValue.value)
```

Comment:
not something we can fix in this PR but making a note here that I think `Value` as a name for this concept is kind of confusing since it doesn't really provide any context. I actually thought it was another name for domain until I got to this line.

I'm wondering if our users also get confused by this. Maybe we can work with Lou and Karthik to come up with a better name.

Reply:
+1 Any variable that ends in `data`, `value`, `object` needs a better name or better abstraction. Likewise any function that starts with `process`, though I've used that to highlight things that need refactoring.

Reply:
Agreed, but definitely not going to try to fix that in this PR

Reply:
https://precisionmedicineinitiative.atlassian.net/secure/RapidBoard.jspa?rapidView=30&projectKey=RW&view=planning&selectedIssue=RW-3596&epics=visible&selectedEpic=RW-3443